third_party/move/tools/move-linter/src/model_ast_lints/assert_const.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module implements an expression linter that checks for assert!()s
+//! where the condition is either `true` or `false`.
+//! Note: As a side-effect, the linter also checks if blocks that are
+//! equivalent to asserts. See the corresponding test file for an example.
+
+use move_compiler_v2::external_checks::ExpChecker;
+use move_model::{
+    ast::{ExpData, Operation, Value},
+    model::FunctionEnv,
+};
+
+#[derive(Default)]
+pub struct AssertConst;
+
+impl ExpChecker for AssertConst {
+    fn get_name(&self) -> String {
+        "assert_const".to_string()
+    }
+
+    fn visit_expr_pre(&mut self, function: &FunctionEnv, expr: &ExpData) {
+        let env = function.env();
+        if let ExpData::IfElse(id, condition, then, else_) = expr {
+            if !Self::is_assert(then, else_) {
+                return;
+            }
+            let condition = Self::get_constant_bool_expression_value(condition);
+            if condition.is_none() {
+                return;
+            }
+            let condition = condition.unwrap();
```

Comment:
Better to avoid unwrap (say by using `if let Some(..)` or similar), or use `expect()` with an appropriate message (for future maintenance and refactoring).